### PR TITLE
Handle lsk

### PR DESCRIPTION
**Pull request by TheoPascoli**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Review by TheMaskedTurtle — CHANGES_REQUESTED** [review]

File: cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/dichotomy/DichotomyRunner.java [hunk 1]
```diff
@@ -77,18 +79,78 @@ private NetworkShifter getNetworkShifter(CseRequest request,
...
-    private ZonalData<Scalable> getZonalScalable(String mergedGlskUrl, Network network) throws IOException {
+    public ZonalData<Scalable> getZonalScalable(String mergedGlskUrl, Network network, ProcessType processType) throws IOException {
         ZonalData<Scalable> zonalScalable = fileImporter.importGlsk(mergedGlskUrl, network);
         Arrays.stream(CseCountry.values()).forEach(country -> checkCseCountryInGlsk(zonalScalable, country));
+        handleLsk(network, zonalScalable, processType);
         return zonalScalable;
     }
 
-    private void checkCseCountryInGlsk(ZonalData<Scalable> zonalScalable, CseCountry country) {
+    private void handleLsk(Network network, ZonalData<Scalable> zonalScalable, ProcessType processType) {
```
Comment (TheMaskedTurtle) [inline cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/dichotomy/DichotomyRunner.java:94]: Maybe rename this method, something like addScalableOnLoads or stackScalableOnLoads

**Review by TheMaskedTurtle — CHANGES_REQUESTED** [review]

File: cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/dichotomy/DichotomyRunner.java [hunk 2]
```diff
@@ -77,18 +79,78 @@ private NetworkShifter getNetworkShifter(CseRequest request,
...
+    }
+
+    private double getZoneSumOfActiveLoads(Network network, String zone) {
+        return network.getLoadStream()
+                    .filter(load -> isLoadCorrespondingToTheZone(load, zone))
+                    .map(Load::getP0)
+                    .mapToDouble(Double::doubleValue).sum();
+    }
+
+    private boolean isLoadCorrespondingToTheZone(Load load, String zone) {
```
[the remark this answers is not in the document]
Comment (TheoPascoli) [reply cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/dichotomy/DichotomyRunner.java:112]: clean

**Review by benrejebmoh — DISMISSED** [review]

File: cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/dichotomy/DichotomyRunner.java [hunk 3]
```diff
@@ -66,29 +68,77 @@ public DichotomyResult<RaoResponse> runDichotomy(CseRequest cseRequest,
...
         Index<RaoResponse> index = new Index<>(MIN_IMPORT_VALUE, MAX_IMPORT_VALUE, dichotomyPrecision);
         DichotomyEngine<RaoResponse> engine = new DichotomyEngine<>(
-            index,
-            new BiDirectionalStepsIndexStrategy(initialIndexValue, initialDichotomyStep),
-            getNetworkShifter(cseRequest, cseData, network),
-            getNetworkValidator(cseRequest, cseData));
+                index,
+                new BiDirectionalStepsIndexStrategy(initialIndexValue, initialDichotomyStep),
+                getNetworkShifter(cseRequest, cseData, network),
+                getNetworkValidator(cseRequest, cseData));
```
Comment (TheMaskedTurtle) [inline cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/dichotomy/DichotomyRunner.java:74]: Please remove the empty spaces here

File: cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/dichotomy/DichotomyRunner.java [hunk 4]
```diff
@@ -66,29 +68,77 @@ public DichotomyResult<RaoResponse> runDichotomy(CseRequest cseRequest,
...
     private NetworkShifter getNetworkShifter(CseRequest request,
                                              CseData cseData,
                                              Network network) throws IOException {
         return new LinearScaler(
-            getZonalScalable(request.getMergedGlskUrl(), network),
-            getShiftDispatcher(request.getProcessType(), cseData, network),
-            SHIFT_TOLERANCE);
+                getZonalScalable(request.getMergedGlskUrl(), network, request.getProcessType()),
+                getShiftDispatcher(request.getProcessType(), cseData, network),
+                SHIFT_TOLERANCE);
```
Comment (TheMaskedTurtle) [inline cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/dichotomy/DichotomyRunner.java:84]: Please remove the empty spaces here

File: cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/dichotomy/DichotomyRunner.java [hunk 5]
```diff
@@ -66,29 +68,77 @@ public DichotomyResult<RaoResponse> runDichotomy(CseRequest cseRequest,
...
+
+        for (Load load : network.getLoads()) {
+            try {
+                if (isLoadCorrespondingToTheZone(load, zone)) {
+                    listPercentages.add((float) (load.getP0() / sum) * 100);
+                    listScalables.add(Scalable.onLoad(load.getId()));
+                }
+            } catch (IllegalArgumentException ignored) {
+                // Catching exception when CseCountry.valueOf() is called for a country not existing, we just ignore it
+            }
```
Comment (TheMaskedTurtle) [inline cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/dichotomy/DichotomyRunner.java:136]: I think now you can remove try/catch block

File: cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/dichotomy/DichotomyRunner.java [hunk 6]
```diff
@@ -97,27 +147,27 @@ private void checkCseCountryInGlsk(ZonalData<Scalable> zonalScalable, CseCountry
     private ShiftDispatcher getShiftDispatcher(ProcessType processType, CseData cseData, Network network) {
         if (processType == ProcessType.D2CC) {
             return new CseD2ccShiftDispatcher(
-                convertSplittingFactors(cseData.getReducedSplittingFactors()),
-                convertBorderExchanges(BorderExchanges.computeCseBordersExchanges(network, true)),
-                convertFlowsOnMerchantLines(cseData.getNtc().getFlowPerCountryOnMerchantLines()));
+                    convertSplittingFactors(cseData.getReducedSplittingFactors()),
+                    convertBorderExchanges(BorderExchanges.computeCseBordersExchanges(network, true)),
+                    convertFlowsOnMerchantLines(cseData.getNtc().getFlowPerCountryOnMerchantLines()));
         } else {
```
Comment (TheMaskedTurtle) [inline cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/dichotomy/DichotomyRunner.java:153]: Please remove the empty spaces here

File: cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/dichotomy/DichotomyRunner.java [hunk 7]
```diff
@@ -97,27 +147,27 @@ private void checkCseCountryInGlsk(ZonalData<Scalable> zonalScalable, CseCountry
...
             return new CseIdccShiftDispatcher(
-                convertSplittingFactors(cseData.getReducedSplittingFactors()),
-                cseData.getCseReferenceExchanges().getExchanges(),
-                cseData.getNtc2().getExchanges());
+                    convertSplittingFactors(cseData.getReducedSplittingFactors()),
+                    cseData.getCseReferenceExchanges().getExchanges(),
+                    cseData.getNtc2().getExchanges());
         }
```
Comment (TheMaskedTurtle) [inline cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/dichotomy/DichotomyRunner.java:158]: Please remove the empty spaces here

File: cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/dichotomy/DichotomyRunner.java [hunk 8]
```diff
@@ -97,27 +147,27 @@ private void checkCseCountryInGlsk(ZonalData<Scalable> zonalScalable, CseCountry
...
-            fileExporter,
-            fileImporter);
+                request.getProcessType(),
+                request.getId(),
+                request.getTargetProcessDateTime(),
+                cseData.getJsonCracUrl(),
+                fileExporter.saveRaoParameters(request.getTargetProcessDateTime(), request.getProcessType()),
+                raoRunnerClient,
+                fileExporter,
+                fileImporter);
```
Comment (TheMaskedTurtle) [inline cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/dichotomy/DichotomyRunner.java:170]: Please remove the empty spaces here

**Review by TheMaskedTurtle — CHANGES_REQUESTED** [review]

**Review by TheMaskedTurtle — APPROVED** [review]

**Pull request merged**